jwala-services/src/main/java/com/cerner/jwala/control/jvm/command/JvmCommandFactory.java

```diff
@@ -184,9 +185,11 @@ private ExecCommand getExecCommandForHeapDump(String scriptName, Jvm jvm) {
 
         final String dumpLiveStr = ApplicationProperties.getAsBoolean(PropertyKeys.JMAP_DUMP_LIVE_ENABLED.name()) ? "live," : "\"\"";
 
-        return new ExecCommand(getFullPathScript(jvm, scriptName), jvm.getJavaHome(),
-                ApplicationProperties.get(PropertyKeys.REMOTE_JAWALA_DATA_DIR), dumpFile, dumpLiveStr, jvmRootDir,
-                jvm.getJvmName());
+        final String heapDumpDir = jvm.getTomcatMedia().getRemoteDir().normalize().toString() + "/" + jvm.getJvmName() +
```

Comment:
Did you test if the heap-dump.sh script will create the heap-dump directory? Might have some permissions issue on Linux